rfcs/0011-project-structure.md

```diff
@@ -0,0 +1,116 @@
+```shell
+├── root
+│   ├── bin/
+│   ├── include/
+│   ├── lib/ # Directory for storing reusable reactors
+│   │   ├── Input.lf # Ex: reactor capturing external inputs (e.g., Microphone, Camera)
+│   │   └── ComputerVision.lf # Ex: reactor performing computer vision tasks (e.g., object detection, face recognition)
+│   ├── src/
+│   ├── src-gen/
+│   ├── fed-gen/              # Only for federated programs (instead of src-gen)
+│   └── Lingo.toml            # Configuration file for Lingo Package Manager
+```
+
+When executing `lingo build`, additional directories will be created in the project root:
+
+```shell
+├── root
+│   ...
+│   ├── target/
+│   │   ├── lfc_include/      # Includes all downloaded external LF projects
+│   │   │   ├── remote_lf_project_1/
+│   │   │   └── remote_lf_project_2/
+│   │   ├── libraries/
+│   │   ├── library/
+│   ...
+```
+
+The ‘Lingua Franca Package Explorer’ in the VS Code extension demonstrates how to utilize this structure. It presents a TreeView with two sections:
+- **Local Libraries**: Lists libraries defined by the programmer, located at `root/lib/*.lf`.
+- **Lingo Libraries**: Lists libraries downloaded from the Lingo Package Manager, located at `root/target/lfc_include/remote_lf_project_1/lib/*.lf`.
+
+This structured approach promotes efficient code reuse, simplifies integration of external resources, and fosters collaborative development practices within the LF community.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Feel free to suggest potential drowbacks of the proposed strucuture.
+
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+### Alternative Names for the `lib/` Folder
+The folder containing reusable reactors has been named `lib/` since the idea is to have a centralized location for all reusable reactors that can be imported into other projects, similar to how libraries are used in other programming languages. However, we’re open to discussing different names for this folder if the proposed one causes confusion among developers. Some alternative names that could be considered include:
+- `reactors_lib/` or `lfc_reactors/`
+- `libraries/`, `lfc_libraries/` or `lfc_lib/`
+- `reusables/`
+- `modules/`
+
+### Alternative Design Approach
+Another approach could involve placing the lib/ folder under the src/ directory, as illustrated below:
+```shell
+├── root
+│   ├── ...
+│   ├── src/
+│   │   ├── lib/ # Directory for storing reusable reactors
+│   │   ├── Main.lf
+│   └── ...
+```
+This structure nests the lib/ folder within src/, organizing reusable reactors alongside other source files. This approach may enhance clarity and organization, but it may also lead to potential conflicts or inconsistencies in the codebase.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+Here are some questions worth considering:
+
+1. Do you think it's beneficial to have a separate `lib/` directory for reusable reactors?
```

Comment:
Yes. I think, however, that there needs to be some clarification on what goes into the `lib/` directory (see my main review comment).